account_tax_balance/hooks.py

```diff
@@ -0,0 +1,74 @@
+# coding: utf-8
+# © 2020 Opener B.V. <https://opener.amsterdam>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+
+def pre_init_hook(cr):
+    """ Precreate move_type and fill with appropriate values to prevent
+    a MemoryError when the ORM attempts to call its compute method on a large
+    amount of preexisting moves. """
+    logger = logging.getLogger(__name__)
+    logger.info('Add account_move.move_type column if it does not yet exist')
+    cr.execute(
+        "ALTER TABLE account_move ADD COLUMN IF NOT EXISTS move_type VARCHAR")
+    cr.execute(
+        """
+        UPDATE account_move am SET move_type = 'liquidity'
+        FROM account_move_line aml
+        WHERE aml.account_id IN (
+            SELECT id FROM account_account
+            WHERE internal_type = 'liquidity')
+        AND aml.move_id = am.id AND am.move_type IS NULL
+        """)
```

Comment:
Well notwithstanding @pedrobaeza comment on a loop to do 5 queries basically the same as this, you could just do a single query with a CASE statement.

Something like
```sql
UPDATE account_move am SET move_type = CASE
  WHEN aa.internal_type = 'liquidity' THEN 'liquidity
  WHEN aa.internal_type = 'payable' and aml.balance < 0 THEN 'payable'
  WHEN aa.internal_type = 'payable' and aml.balance >= 0 THEN 'payable_refund'
  WHEN aa.internal_type = 'receivable' and aml.balance > 0 THEN 'receivable'
  WHEN aa.internal_type = 'receivable' and aml.balance <= 0 THEN 'receivable_refund'
  ELSE 'other' 
END
FROM account_move_line aml, account_account aa
WHERE aa.id = aml.account_id 
      AND aml.move_id = am.id AND am.move_type IS NULL
```

Although you'd lose the rowcount info for logging, it is just one query and self explanatory.

Reply:
Alright, thanks both. I'll go with @gdgellatly's version then.